auth/src/user.rs

```diff
@@ -0,0 +1,170 @@
+use std::{fmt::Formatter, str::FromStr};
+
+use async_trait::async_trait;
+use serde::{Deserialize, Deserializer, Serialize};
+use sqlx::{query, Pool, Row, Sqlite};
+
+use crate::error::Error;
+
+#[async_trait]
+pub(crate) trait UserManagement {
+    async fn create_user(&self, name: AccountName) -> Result<User, Error>;
+    async fn get_user(&self, name: AccountName) -> Result<User, Error>;
+}
+
+#[derive(Clone)]
+pub(crate) struct UserManager {
+    pub(crate) pool: Pool<Sqlite>,
+}
+
+#[async_trait]
+impl UserManagement for UserManager {
+    async fn create_user(&self, name: AccountName) -> Result<User, Error> {
+        // TODO: generate a secret
+        let secret = "my_secret".to_owned();
+
+        query("INSERT INTO users (account_name, secret) VALUES (?1, ?2)")
+            .bind(&name)
+            .bind(&secret)
+            .execute(&self.pool)
+            .await?;
+
+        Ok(User::new_with_defaults(name, secret))
+    }
+
+    // TODO: get from token?
```

Comment:
Nah, the account should be known at this level. Only the handlers should know about tokens.

For the bigger use case of adding a user the details are as follow: some admin account will post the account to add to `/user/:user_name`

Reply:
Cool, thanks!